functorch/_src/aot_autograd.py

```diff
@@ -286,7 +286,58 @@ def new_fn(args):
 
 
 def aot_dispatch_autograd(flat_fn, flat_args: List[Tensor], aot_config: AOTConfig):
-    joint_forward_backward = create_joint_forward_backward(flat_fn)
+    # Deduplicate inputs.  Suppose you have:
```

Comment:
I don't think this should be done here. It should be done either in torchdynamo, or in `aot_function` (or `create_aot_dispatcher_function`) imo.

Reply:
If you do it in `aot_function` you will miss the analogous bug with modules. create_aot_dispatcher_function is plausible, but you technically only need to do this in the autograd case so... meh?

See PR comment about torchdynamo.